src/boot/efi/meson.build

```diff
@@ -400,6 +413,32 @@ stub_sources = files(
         'stub.c',
 )
 
+if efi_zstd != ''
+        efi_cflags += [
+                '-isystem' + efi_zstd,
+                '-DDEBUGLEVEL=0',
+                '-DXXH_DEBUGLEVEL=0',
+                '-DDYNAMIC_BMI2=0',
+                '-DZSTD_DISABLE_ASM',
+                '-DZSTD_LEGACY_SUPPORT=0',
+                '-DZSTD_NO_INTRINSICS',
+                '-DZSTD_NO_UNUSED_FUNCTIONS',
+        ]
+
+        stub_sources += files(
+                efi_zstd / 'common/debug.c',
+                efi_zstd / 'common/entropy_common.c',
+                efi_zstd / 'common/error_private.c',
+                efi_zstd / 'common/fse_decompress.c',
+                efi_zstd / 'common/xxhash.c',
+                efi_zstd / 'common/zstd_common.c',
+                efi_zstd / 'decompress/huf_decompress.c',
+                efi_zstd / 'decompress/zstd_ddict.c',
+                efi_zstd / 'decompress/zstd_decompress_block.c',
+                efi_zstd / 'decompress/zstd_decompress.c',
+        )
+endif
```

Comment:
i wished they'd simply provide a meson subproject thingy, that would allow us to let them maintain this list, not us.

Reply:
They do have one, but it doesn't include support for the freestanding bits I guess.